controllers/ibmblockcsi_controller.go

```diff
@@ -201,40 +203,60 @@ func (r *IBMBlockCSIReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	return reconcile.Result{}, nil
 }
 
+// SetupWithManager sets up the controller with the Manager.
+func (r *IBMBlockCSIReconciler) SetupWithManager(mgr ctrl.Manager) error {
+
+	serverVersion, err := getServerVersion()
+	if err != nil {
+		panic(err)
+	}
+
+	log.Info(fmt.Sprintf("Kubernetes Version: %s", serverVersion))
+
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&csiv1.IBMBlockCSI{}).
+		Owns(&appsv1.StatefulSet{}).
+		Owns(&appsv1.DaemonSet{}).
+		Owns(&corev1.ServiceAccount{}).
+		Complete(r)
+}
+
 func getServerVersion() (string, error) {
 	kubeVersion, found := os.LookupEnv(oconfig.ENVKubeVersion)
 	if found {
 		return kubeVersion, nil
 	}
 
-	kubeClient := kubeutil.KubeClient
+	clientConfig, err := GetClientConfig()
+	if err != nil {
+		return "", err
+	}
+
+	kubeClient := kubeutil.InitKubeClient(clientConfig)
 
-	serverVersion, err := kubeutil.ServerVersion(kubeClient.Discovery())
+	serverVersion, err := serverVersion(kubeClient.Discovery())
 	if err != nil {
 		return serverVersion, err
 	}
-	if strings.HasSuffix(serverVersion, "+") {
-		serverVersion = strings.TrimSuffix(serverVersion, "+")
-	}
+	serverVersion = strings.TrimPrefix(serverVersion, "+")
```

Comment:
why this change? could you give an example for a server version value?

Reply:
not sure I follow.
why do we even need to trim anything?

Reply:
I don't know, we needed trim even before my change, probably we get the Kubernetes version with `+` in the string and we wanted to remove them

Reply:
I don't think we needed trim. I don't see where we could get a `+`, and even if we did, I don't think your change would remove it.
please remove the trimming code altogether

Reply:
Do you want me also remove the old code?